src/taipy/core/scenario/_scenario_manager.py

```diff
@@ -203,8 +203,13 @@ def _untag(cls, scenario: Scenario, tag: str):
 
     @classmethod
     def _delete(cls, scenario_id: ScenarioId):  # type: ignore
-        if cls._get(scenario_id).is_primary:
-            raise DeletingPrimaryScenario
+        scenario = cls._get(scenario_id)
+        if scenario.is_primary:
+            if len(cls._get_all_by_cycle(scenario.cycle)) > 1:
```

Comment:
I think you can wrap this as another function (delete_primary_scenario) to reduce duplicate code with the lines in _hard_delete